evm_arithmetization/src/cpu/kernel/asm/core/jumpdest_analysis.asm

```diff
@@ -157,137 +153,50 @@ global write_table_if_jumpdest:
-    PUSH 0xFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF000000000000000000
-    AND
-    // stack: (is_0_at_5|X⁷)⁷|(0⁸)⁸|(is_0_at_5|X⁷)⁸|(0⁸)⁸, (is_0_at_5|X⁷)³²,  (in_range|X⁷)³², (~has_prefix|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-    DUP4
-    OR
-    // stack: (~has_prefix'|X⁷)³², (is_0_at_5|X⁷)³²,  (in_range|X⁷)³², (~has_prefix|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-    SWAP3
-    OR
-    // pos 0102030405060708091011121314151617181920212223242526272829303132
-    PUSH 0xFFFFFFFFFFFFFF0000000000000000FFFFFFFFFFFFFFFF000000000000000000
-    OR
-    AND
-    // stack: (in_range'|X⁷)³²,  (~has_prefix'|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-
-    // Compute in_range' and ~has_prefix' as
-    //   - in_range' = in_range                                     and ~has_prefix' = ~has_prefix OR is_0_at_6, for bytes in positions 1-3, 8-11, 16-19, and 24-27 
-    //   - in_range' = in_range AND (has_prefix => is_0_at_6 |X⁷)³² and ~has_prefix' = has_prefix,               for the rest.
-    DUP3 %shl_const(5)
-    NOT
-    // stack: (is_0_at_6|X⁷)³²,  (in_range|X⁷)³², (~has_prefix|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-    DUP1
-    // pos 0102030405060708091011121314151617181920212223242526272829303132
-    PUSH 0xFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF0000000000
-    AND
-    // stack: (is_0_at_6|X⁷)³|(0⁸)⁴|((is_0_at_6|X⁷)⁴|(0⁸)⁴)³, (is_0_at_6|X⁷)³²,  (in_range|X⁷)³², (~has_prefix|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-    DUP4
-    OR
-    // stack: (~has_prefix'|X⁷)³², (is_0_at_6|X⁷)³²,  (in_range|X⁷)³², (~has_prefix|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
-    SWAP3
-    OR
-    // pos 0102030405060708091011121314151617181920212223242526272829303132
-    PUSH 0xFFFFFF00000000FFFFFFFF00000000FFFFFFFF00000000FFFFFFFF0000000000
-    OR
+    %and_const(0x1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F1F)
+    // stack: (000|X⁵)³², mask, (is_push|X⁷)³², packed_opcodes, proof_prefix_addr, ctx, jumpdest, retdest
+    // For opcode PUSHXX, the right-most 5 bits contain XX - 1.
+    // Ignoring the first 3 bits of prefix, the first opcode must NOT be PUSH32, the second opcode
+    // must NOT be PUSH31 or PUSH32 [...], the 32-th opcode must NOT be a PUSH.
+    // We can check it by adding the trimmed opcodes with a certain value such that the addition overflows iff
+    // the five bits of an opcode are forbidden:
+    //     000xxxxx|000xxxxx|...|000xxxxx|000xxxxx
+    //   + 00000001|00000010|...|00011111|00100000
+    // For e.g. the first opcode, the addition will overflow iff xxxxx = 0b11111 = 0d31.
+    // For the last opcode, since any PUSHXX operation is forbidden, the overflow bit is set manually.
+    // Note that since a five-bit addition will always use at most six bits, the overflow bit will always be
```

Comment:
```suggestion
    // Note that since the result of a five-bit addition will always use at most six bits, the overflow bit will always be
```

Reply:
Except for the last position where you add `0x20 = 32`, a 6-bit number. Luckily a 32 + any 5-bit number is still a 6-bit number.